### PR TITLE
[Re-arch]: List Files Ability

### PR DESCRIPTION
**Pull request by ntindle**
### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes. 
- [ ] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```

### CODE REVIEW

**Comment by ntindle** [comment]
@collijk One thing I noticed is `data` isn't actually part of an ability result. Wanted to point that our

**Comment by ntindle** [comment]
@collijk Do we want to restrict to workspace only?

**Comment by ntindle** [comment]
related in #4770

File: autogpt/core/ability/builtins/file_operations.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def __call__(self, filename: str) -> AbilityResult:
 
 class WriteFile(Ability):
     default_configuration = AbilityConfiguration(
-        packages_required=["unstructured"],
+        packages_required=["os"],
```
Comment (collijk) [inline autogpt/core/ability/builtins/file_operations.py:87]: We don't need builtin packages here.

File: autogpt/core/ability/builtins/file_operations.py [hunk 2]
```diff
@@ -155,3 +155,80 @@ def __call__(self, filename: str, contents: str) -> AbilityResult:
             message=message,
             data=data,
         )
+
+
+class ListFiles(Ability):
+    default_configuration = AbilityConfiguration(
+        packages_required=["os"],
```
Comment (collijk) [inline autogpt/core/ability/builtins/file_operations.py:162]: Same

**Comment by collijk** [comment]
> @collijk One thing I noticed is `data` isn't actually part of an ability result. Wanted to point that out

Feel free to add.  I haven't dug into knowledge summarization yet so haven't had to use it.

**Comment by collijk** [comment]
> @collijk Do we want to restrict to workspace only?

The workspace should be managing that (and it's user configurable)

**Comment by Pwuts** [comment]
Branch deleted for security reasons

**Pull request closed without merging**